jscomp/others/js_global.ml

```diff
@@ -0,0 +1,34 @@
+type intervalId
+type timeoutId
+(* part of WindowOrWorkerGlobalScope, to be implemented later
+type cacheStorage (** Service Workers, not yet implemented *)
+type idbFactory (** Indexed Database API, not yet implemented *)
+type imageBitmap (** not yet implemented *)
+type ('a, 'e) promise (* should this type be defined in Js, and be aliased by bs-promise and other implementations? *)
+*)
+
+(* part of WindowOrWorkerGlobalScope, to be implemented later
+external caches : cacheStorage = "" [@@bs.val] (* Service Workers, experimental *)
+external indexedDB : idbFactory = "" [@@bs.val] (* Indexed Database API *)
+external isSecureContext : bool = "" [@@bs.val] (* Secure Contexts, experimental *)
+external origin : string = "" [@@bs.val] (* WHATWG HTML Living Standard, experimental *)
+*)
+
+(** convert an ASCII-encoded string to its Base64-encoded equivalent *)
+external atob : string -> string = "" [@@bs.val]
+
+(** convert a Base64-encoded string to its ASCII-encoded equivalent *)
+external btoa : string -> string = "" [@@bs.val]
+
+external clearInterval : intervalId -> unit = "" [@@bs.val]
```

Comment:
Just to clarify. I consider `window`-specific functions those that depend on a specific instance of a window, and which in bs code will need to be explicitly passed a window instance. `Global` are those functions that do not depend on a specific window instance, but that are attached to `window` merely because it for silly historic reasons happens to be the global scope.